hpack/table.py

```diff
@@ -172,12 +185,15 @@ def search(self, name, value):
         """
         offset = HeaderTable.STATIC_TABLE_LENGTH + 1
         partial = None
-        for (i, (n, v)) in enumerate(HeaderTable.STATIC_TABLE):
-            if n == name:
-                if v == value:
-                    return i + 1, n, v
-                elif partial is None:
-                    partial = (i + 1, n, None)
+
+        header_name_search_result = HeaderTable.STATIC_TABLE_MAPPING.get(name)
+        if header_name_search_result is not None:
+            index = header_name_search_result.positions.get(value)
+            if index is not None:
+                return index + 1, name, value
+            else:
+                partial = (header_name_search_result.min + 1, name, None)
```

Comment:
dict search: 
```python
In [26]: %timeit ht.search(b':authority', b'')
The slowest run took 12.41 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 557 ns per loop

In [27]: %timeit ht.search(b'www-authenticate', b'')
The slowest run took 13.68 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 593 ns per loop
```
linear search:
```python
In [31]: %timeit ht.search(b':authority', b'')
The slowest run took 11.18 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 618 ns per loop

In [32]: %timeit ht.search(b'www-authenticate', b'')
The slowest run took 9.54 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 4.2 µs per loop
```